src/System.Runtime.Serialization.Formatters/tests/EqualityExtensions.cs

```diff
@@ -903,6 +903,27 @@ public static bool IsEqual(this Version @this, Version other)
                 @this.MinorRevision == other.MinorRevision;
         }
 
+        public static bool IsEqual(this Exception @this, Exception other)
+        {
+            return @this != null &&
+                other != null &&
+                // On full framework, line number may be method body start
+                (PlatformDetection.IsFullFramework ? true :
+                (@this.StackTrace == other.StackTrace &&
+                @this.ToString() == other.ToString())) &&
```

Comment:
Strictly one would not require ToString() to match, but I see it's the only public way to verify ClassName.

Reply:
if it turns out they produce different strings, you may just have to do IndexOf for ClassName in the string..